Sources/TUSKit/TUSAPI.swift

```diff
@@ -436,7 +444,11 @@ extension Dictionary {
 private extension TUSAPI {
     final class SessionDataDelegate: NSObject, URLSessionDataDelegate {
         weak var api: TUSAPI?
-        
+
+        func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive data: Data) {
+            api?.handleCompletionOfTask(dataTask, withError: nil, data: data)
```

Comment:
This method being called shouldn't be considered completion of the request. It can be called multiple times with partial responses. We should store the received data somewhere in a dictionary of `[URLsessionDataTask: Data]` or similar so that we can append received data when needed and we can grab the correct data per task in `didCompleteWithError`

See docs here: https://developer.apple.com/documentation/foundation/urlsessiondatadelegate/urlsession(_:datatask:didreceive:)

Reply:
That's a good call - can make that change 👍

Reply:
Updated here [f6e7c03](https://github.com/tus/TUSKit/pull/213/commits/f6e7c030f8f59c83c76d7f47776f31416bffc27f).